tutorials/convert_lit_models.md

```diff
@@ -1,23 +1,118 @@
-## Converting Lit-GPT weights to HuggingFace Transformers
+## Converting Lit-GPT weights to Hugging Face Transformers
 
-Lit-GPT weights need to be converted to a format that HuggingFace understands with a [conversion script](../scripts/convert_lit_checkpoint.py) before our scripts can run.
+Lit-GPT weights need to be converted to a format that Hugging Face understands with a [conversion script](../scripts/convert_lit_checkpoint.py) before our scripts can run.
 
-We provide a helpful script to convert models Lit-GPT models back to their equivalent HuggingFace Transformers format:
+We provide a helpful script to convert models Lit-GPT models back to their equivalent Hugging Face Transformers format:
 
 ```sh
 python scripts/convert_lit_checkpoint.py \
     --checkpoint_path path/to/litgpt/model.pth \
-    --output_path where/to/save/the/converted.ckpt \
+    --output_path out/converted.ckpt \
     --config_path path/to/litgpt/config.json
 ```
 
 These paths are just placeholders, you will need to customize them based on which finetuning or pretraining script you ran and it's configuration.
 
+### Loading converted Lit-GPT checkpoints into transformers
+
+If you want to load the converted checkpoints into a `transformers` model, please make sure you copied the original `config.json` file into the folder that contains the `converted.ckpt` file saved via `--output_path` above.
+
+For example,
+
+```bash
+wget https://huggingface.co/repo_id/raw/main/config.json
+mv config.json where/to/save/the/
+```
+
+Then, you can load the checkpoint file in a Python session as follows:
+
+```python
+import torch
+from transformers import AutoModel
+
+
+state_dict = torch.load("out/converted.ckpt")
+model = AutoModel.from_pretrained(
+    "out/", local_files_only=True, state_dict=state_dict
+)
+```
+
+Alternatively, you can also load the model without copying the `config.json` file as follows:
+
+```python
+model = AutoModel.from_pretrained("online_repo_id", state_dict=state_dict)
+```
+
+
+
+### Merging LoRA weights
+
 Please note that if you want to convert a model that has been fine-tuned using an adapter like LoRA, these weights should be [merged](../scripts/merge_lora.py) to the checkpoint prior to converting.
 
 ```sh
 python scripts/merge_lora.py \
-    --checkpoint_dir path/to/litgpt/model.pth \
+    --checkpoint_dir path/to/litgpt_checkpoint/ \
     --lora_path path/to/litgpt/lora_finetuned.pth \
     --out_dir where/to/save/the/merged.ckpt
 ```
+
+<br>
+<br>
+
+# A finetunig and conversion tutorial
+
+This section contains a reproducible example for finetuning a Lit-GPT model and converting it back into a HF `transformer` model.
+
+1) Download a model of interest:
+
+```bash
+python scripts/download.py \
+    --repo_id TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T
```

Comment:
I personally prefer to export the name into a variable and then reuse it in all the required scripts.
```bash
export repo_id=TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T
python scripts/download.py --repo_id $repo_id
python scripts/convert_hf_checkpoint.py --checkpoint_dir checkpoints/$repo_id
...
```

Reply:
That's actually a good idea, I just wonder if this generalizes well to different shell environments? (Bash, zsh, fish, etc.)

Reply:
Just validated in bash, zsh and fish (all the latest versions).
Worked out without problems.

But this was a good question.
Previously fish didn't support exporting value, you had to use their syntax:
```bash
set -Ux repo_id TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T
```

but now it supports both.